src/db/collections/block.rs

```diff
@@ -264,23 +265,35 @@ impl BlockCollection {
         &self,
         index: MilestoneIndex,
     ) -> Result<impl Stream<Item = Result<(BlockId, Block, Vec<u8>, BlockMetadata), Error>>, Error> {
+        #[derive(Debug, Deserialize)]
+        struct QueryRes {
+            #[serde(rename = "_id")]
+            block_id: BlockId,
+            #[serde(with = "serde_bytes")]
+            raw: Vec<u8>,
+            metadata: BlockMetadata,
+        }
```

Comment:
Why is this type necessary? Could we not just ignore `block` when mapping to our returned tuple? Or is this already the problem that the deserialization will fail?

Reply:
Deserialization will fail. We should consider removing `derive(Deserialize)`.

Reply:
It's unfortunate but we are allowed to gather the outputs by lookup, so it still needs to be `Deserialize`.